### ISSUE TITLE
Referenced host+subset in destinationrule not found: "hostname+version-v3"

### ISSUE THREAD

**Issue by pseudonator** [istio/istio#36237]
### Bug Description

Appears that `istioctl analyze` will fail due to the following error,
```
Error [IST0101] (VirtualService reviews.default) Referenced host+subset in destinationrule not found: "hostname+version-v3"
Error: Analyzers found issues when analyzing namespace: default.
See https://istio.io/v1.11/docs/reference/config/analysis for more information about causes and resolutions.
```
when `DestinationRule` is written to the root config namespace (`istio-system`) and `VirtualService` is written to an entirely different namespace (`default` in this case) and subsets are used.

### Version

Istio version,
```prose
❯ istioctl version
client version: 1.11.4
control plane version: 1.11.4
data plane version: 1.11.4 (6 proxies)
```

### Additional Information

Digging into the code it seems like analyzer https://github.com/istio/istio/blob/master/pkg/config/analysis/analyzers/virtualservice/destinationrules.go#L95 is restricting it self to the namespace where the `VirtualService` is. This should really be looking into the root namespace.

**Comment by pseudonator** [comment]
@nrjpoddar As discussed heres the bug around the analyzer.

**Comment by nrjpoddar** [comment]
@pseudonator thanks, it might be good to add the exact VS and DR (after scrubbing) so folks can use it for their test case and reproducing the issue locally.

**Comment by zhlsunshine** [comment]
hi @pseudonator & @nrjpoddar ,  so which are that DestinationRule should be matched with below?
1. VirtualService's namespace
2. the DestinationRule's namespace
3. The namespace of host which is belong to specified VirtualService
4. Root namespace which Istio has been installed
1 has been considered for now, so according to @pseudonator's comment, 1 and 2 should be considered together. Am I correct?

**Comment by nrjpoddar** [comment]
1. Client namespace i.e. VirtualService's
2. Server namespace derived from FQDN of the destination host
3. Root config namespace

**Comment by zhlsunshine** [comment]
Hi @pseudonator and @nrjpoddar,  there is new comment for the PR [here](https://github.com/istio/istio/pull/36402#discussion_r764037691).  And then, there should be still the error message, but it's different with the original one.

**Comment by zhlsunshine** [comment]
Hi @howardjohn , @pseudonator and @nrjpoddar,  I am working on this issue in [PR#36402](https://github.com/istio/istio/pull/36402), after long discussion, finally I find out that it should be an invalidated issue if there is no special handling for destination rule belonged to root namespace. 
As description in this issue, the destination rule in `istio-system and VirtualService in `default`, moreover, both DR and VS have short name host with specified subsets. Then, host in DR has been interpreted as `reviews.istio-system.svc.cluster.local` and host in VS has been interpreted as `reviews.default.svc.cluster.local`. So there should be an error message when doing analysis.
And @pseudonator has mentioned the code here: https://github.com/istio/istio/blob/master/pkg/config/analysis/analyzers/virtualservice/destinationrules.go#L95, in fact, it should be okay because this analyzer need to verify that the destination defined in VS is declared in DR.

**Comment by zhlsunshine** [comment]
Hi @howardjohn , @pseudonator and @nrjpoddar,  I am working on this issue in [PR#36402](https://github.com/istio/istio/pull/36402), after long discussion, finally I find out that it should be an invalidated issue if there is no special handling for destination rule belonged to root namespace. 
As description in this issue, the destination rule in `istio-system` and VirtualService in `default`, moreover, both DR and VS have short name host with specified subsets. Then, host in DR has been interpreted as `reviews.istio-system.svc.cluster.local` and host in VS has been interpreted as `reviews.default.svc.cluster.local`. So there should be an error message when doing analysis.
And @pseudonator has mentioned the code here: https://github.com/istio/istio/blob/master/pkg/config/analysis/analyzers/virtualservice/destinationrules.go#L95, in fact, it should be okay because this analyzer need to verify that the destination defined in VS is declared in DR.

**Comment by nrjpoddar** [comment]
Thanks @zhlsunshine for being patient and working through the feedback. So the real issue is the use of shortnames in the DR in istio-system namespace. @pseudonator can you verify that using a FQDN fixes the issue?

**Comment by zhlsunshine** [comment]
Thanks @nrjpoddar, and for short name usage in the DR in istio-system namespace, I think @howardjohn has try it and works fine, and I understand that we may treat the root namespace as a normal one for DR.  Any idea? @pseudonator and @nrjpoddar

**Comment by pseudonator** [comment]
@zhlsunshine I propose we close this ticket and you already have an ongoing PR.

tldr; Reason I say this is because I can no longer reproduce this in a single cluster. The analyzer still fails across clusters but I believe thats expected. For instance if I analyze `cluster-1` resources and there is VS with a FQDN `reviews.bookinfo.svc.cluster-2.global` which points to `cluster-2` its going to fail resolving that.

Feel free to re-open if you feel we need more context around this.
